app/models/katello/concerns/redhat_extensions.rb

```diff
@@ -72,8 +72,8 @@ def distribution_repositories(host)
 
         if content_view && lifecycle_environment && host.os && host.architecture
           Katello::Repository.in_environment(lifecycle_environment).in_content_views([content_view]).
-              where(:distribution_version => host.os.release,
-                    :distribution_arch => host.architecture.name)
+              where(:distribution_arch => host.architecture.name).
+              where("#{Katello::Repository.table_name}.distribution_version like ?", "#{host.os.release}%")
```

Comment:
wouldn't this cause '7.1' to match '7.10' ?

Reply:
updated